README.md

```diff
@@ -83,8 +83,8 @@ export default {
     return {
       baz: 10
       ...this.mapState({
-        baz: function (state) {
-          return state.bar + this.baz //maps state.bar + 10 to data.baz
+        foo: function (state) {
+          return state.bar + this.baz //maps state.bar + 10 to data.foo
```

Comment:
Although `this` in the `foo` function is bound correctly to the Vue component, `this.baz` unfortunately comes up as `undefined`.

I think this is because the `data` properties are not yet available in the `beforeCreate` lifecycle-hook and only become available in the `created` hook.  

I tried adding `this.mapState` in the `created` hook but, as expected, this breaks immediately because the data instantiation expects `this.mapState` to already be available.

It might even be inherently impossible to have access to `this.baz` when the `data` object is instantiated, hence why Vuex's `mapState` is maybe mapped to the `computed` properties object instead of the `data` object (in order to have access to the `data` object reactively). I might be wrong tho.

Wrapping the `console.log` in a `setTimeout` callback in order to push the logging to the next ticks successfully logs `this.baz`.

```js
  data() {
    return {
      baz: 10,
      ...this.mapState({
        foo: function(state) {
          setTimeout(() => console.log(this.baz), 0);
          return state.bar + this.baz //maps state.bar + 10 to data.foo
        }
      })
    };
  }

// CONSOLE => 10
```

Personally, I could do without having access to the `data` properties  in `mapState` if this is not easy to resolve, because I could always create a `computed` property that combines the two states like so :)


```js
{
  data() {
    return {
      baz: 10,
      ...this.mapState({
        bar: function(state) {
          return state.bar
        }
      })
    };
  },
  computed: {
    foo() {
      return this.baz + this.bar;
    }
  }
}
```

Reply:
Yes, actually I was aware of this issue. I tried hard to get this working but because of the vue component lifecycle I would have to re evaluate all state bindings in the created hook, which doubles the memory imprint :/ I think I might remove this example to not confuse consumers.

Reply:
That's what I thought. Honestly this library is already amazing without this feature so I don't see it as a problem, especially considering the memory cost it would incur.